src/lib/cfa/ntp_conf.rb

```diff
@@ -179,24 +179,23 @@ class Record
       # to its AugeasElement key.
       # @param [String] key
       def self.new_from_augeas(augeas_entry)
-        record = record_class(augeas_entry[:key]).new
-        record.augeas = augeas_entry
-        record
+        record_class(augeas_entry[:key]).new(augeas_entry)
       end
 
       # Returns the corresponding subclass
       # @param [string] key
       def self.record_class(key)
         entry_type = key.gsub("[]", "")
-        record_class = ["::CFA::NtpConf::", entry_type.capitalize, "Record"].join
+        record_class = "::CFA::NtpConf::#{entry_type.capitalize}Record"
         Kernel.const_get(record_class)
       end
 
-      def initialize(key = nil)
-        @augeas = { key: key, value: nil }
+      def initialize(augeas = nil)
+        augeas ||= create_augeas
+        @augeas = augeas
```

Comment:
good, now I see how it can act on augeas tree as you pass now reference to it. Previously it was well hidden.